liblocation/src/main/java/com/mapbox/android/core/location/AndroidLocationEngineImpl.java

```diff
@@ -141,16 +89,19 @@ public void removeLocationUpdates(@NonNull LocationListener listener) {
   }
 
   @Override
-  public void removeLocationUpdates(@NonNull PendingIntent pendingIntent) {
+  public void removeLocationUpdates(PendingIntent pendingIntent) {
     if (pendingIntent != null) {
       locationManager.removeUpdates(pendingIntent);
     }
   }
 
-  @VisibleForTesting
-  @Override
-  public int getListenersCount() {
-    return registeredListeners();
+  private String getBestProvider(int priority) {
+    String provider = null;
```

Comment:
well, `locationManager.getBestProvider` can return null pretty frequently even if you assign a valid default value (null initialization is slightly cheaper), it can be overwritten by `null` value as a result, so you must check for `null`!

Reply:
I wasn't aware `locationManager.getBestProvider` can return `null` :+1: